### DISCUSSION TITLE
Getting the body from a ServiceResponse

### DISCUSSION THREAD

**Discussion by joepio** [actix/actix-web#2459] [General]
Hi! I'm upgrading my actix-web graph database server to `4.0.0-beta.12`, and I noticed that the `AnyBody` type has been [introduced](https://github.com/actix/actix-web/pull/2448). This changes the previous API for accessing the body from a response in tests.

This is what I used to do:

```rust
    let mut resp = test::call_service(&mut app, req).await;
    let body = resp.take_body();
    assert!(body.as_str().contains("html"));
````

But that no longer works, I now need to do something like:

```rust
    let mut resp = test::call_service(&mut app, req).await;
    let body = match resp.into_body() {
        AnyBody::None => panic!("no string body"),
        AnyBody::Bytes(b) => String::from_utf8(b.as_ref().into()).unwrap(),
        AnyBody::Body(_) => panic!("no string body"),
    };
    assert!(body.contains("html"));
```

Maybe I'm missing something, but getting the string representation of a response body shouldn't be this difficult, right?

**Comment by fakeshadow** [comment]
```rust
async fn test(res: actix_web::dev::ServiceResponse) {
    let bytes = actix_http::body::to_bytes(res.into_body()).await.unwrap();
    assert!(std::str::from_utf8(&bytes).unwrap().contains("html"));
}
```

**Reply by joepio** [reply]
Thanks for sharing. That indeed is a bit shorter, but it's still far too verbose in my taste, since the usecase (getting body text in tests) is so incredibly common.
